Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnUpdate.hs

```diff
@@ -243,22 +251,25 @@ onUpdate RideCompletedReq {..} = do
           }
 onUpdate BookingCancelledReq {..} = do
   booking <- QRB.findByBPPBookingId bppBookingId >>= fromMaybeM (BookingDoesNotExist $ "BppBookingId: " <> bppBookingId.getId)
-  unless (isBookingCancellable booking) $
-    throwError (BookingInvalidStatus (show booking.status))
   mbRide <- QRide.findActiveByRBId booking.id
+
+  let isBookingCancellable =
+        booking.status `elem` [SRB.NEW, SRB.CONFIRMED, SRB.AWAITING_REASSIGNMENT, SRB.TRIP_ASSIGNED]
+      isRideCancellable = maybe False (\ride -> ride.status /= SRide.CANCELLED) mbRide
```

Comment:
@olegromashin Even if ride status is INPROGRESS, then also we can't cancel the ride.

Reply:
Done, please check